src/go/languageservice/workspace.go

```diff
@@ -169,20 +171,54 @@ func (id WorkspaceID) GetWorkspaceErrors() []error {
 	return errors
 }
 
-func (id WorkspaceID) GetCompletions() ([]string, error) {
+func (id WorkspaceID) GetCompletions(position int) ([]string, error) {
 
 	workspace, err := id.getWorkspace()
 	if err != nil {
 		return nil, *err
 	}
 
 	completions := []string(nil)
-
-	// TODO: take into account context.
-	// TODO: support locals.
+    
+	// TODO: support other node types.
 	// TODO: return item type.
 	// TODO: support fetching item description.
 	for _, wd := range workspace.Files {
+        codePos := token.Pos(position)
+        // current position: =  wd.FileSet.Position(codePos).String()
+        enclosingPath, _ := astutil.PathEnclosingInterval(wd.File, codePos, codePos+1)
+	    
+        // locals form enslosing scopes, currently only function declarations:
+	    for _ , enclosingNode := range(enclosingPath) {
+		    switch x := enclosingNode.(type) {
+
+                // for function declarations, get all their locals declared before codePos 
+		        case *ast.FuncDecl:
```

Comment:
Does this still work with top level constructs like functions, types, vars, etc?

Reply:
Currently, I am handling only functions.